chia/server/start_service.py

```diff
@@ -163,19 +163,16 @@ async def start(self, **kwargs) -> None:
         self._rpc_close_task = None
         if self._rpc_info:
             rpc_api, rpc_port = self._rpc_info
-            self._rpc_task = asyncio.create_task(
-                start_rpc_server(
-                    rpc_api(self._node),
-                    self.self_hostname,
-                    self.daemon_port,
-                    uint16(rpc_port),
-                    self.stop,
-                    self.root_path,
-                    self.config,
-                    self._connect_to_daemon,
-                    max_request_body_size=self.max_request_body_size,
-                    name=self._service_name + "_rpc",
-                )
+            self.rpc_server = await start_rpc_server(
```

Comment:
This is a change in behaviour. Before we were not waiting for the rpc server to actually get created, now we are. Do you think this will cause some issues? I don't exactly remember why we did it like this, but I remember there were some issues in this area of the code

Reply:
Perhaps this might even fix the flaky issue?

Reply:
> This is a change in behaviour. Before we were not waiting for the rpc server to actually get created, now we are. Do you think this will cause some issues?

Right, i thought about it when i made the change but i didn't see a reason why it would be an issue. 

> I don't exactly remember why we did it like this, but I remember there were some issues in this area of the code

We are doing it the same way already for the other API's in `ChiaServer.start_server` so i guess its just fine? 

> Perhaps this might even fix the flaky issue?

Yeah maybe, you have your ways of reproducing it, right? Maybe give it a try with this branch :)